### ISSUE TITLE
[Feature Request] Google Safe Browsing

### ISSUE THREAD

**Issue by ai-utt** [AdguardTeam/AdGuardHome#1834]
- what is Google Safe Browsing:
Safe Browsing is a Google service that lets client applications check URLs against Google's constantly updated lists of unsafe web resources. Examples of unsafe web resources are social engineering sites (phishing and deceptive sites) and sites that host malware or unwanted software.
 
i know it's already been requested #1446 
but is there any chance that this can be implemented in v0.103? 

thank you.

**Comment by mHatsune** [comment]
https://github.com/AdguardTeam/AdGuardHome/issues/1446#issuecomment-649572082

**Comment by ameshkov** [comment]
> but is there any chance that this can be implemented in v0.103?

No, we have too much planned for v0.103 and v0.104, I am quite sure we won't be able to fit this one.

Also, it depends on how many users request this feature.

**Comment by mHatsune** [comment]
> i know but it doesn't use google safe browsing, sadly Cloudflare malware blocking, quad9 and many other “malware, phishing etc...” blocking DNSs are blocked in my country.

What country are you from?

**Comment by SanderGit** [comment]
As a workaround, could you add your own [DNS rewrite](https://support.opendns.com/hc/en-us/articles/227986807-How-to-Enforcing-Google-SafeSearch-YouTube-and-Bing)?

**Comment by ai-utt** [comment]
> > i know but it doesn't use google safe browsing, sadly Cloudflare malware blocking, quad9 and many other “malware, phishing etc...” blocking DNSs are blocked in my country.
> 
> What country are you from?

Saudi Arabia.

**Comment by ai-utt** [comment]
> As a workaround, could you add your own [DNS rewrite](https://support.opendns.com/hc/en-us/articles/227986807-How-to-Enforcing-Google-SafeSearch-YouTube-and-Bing)?

Thank you, but i meant [Google SafeBrowsing](https://safebrowsing.google.com) and not Google SafeSearch, tbf they both sounds really similar. and actually Google SafeSearch is already enabled and option locked [ON] by default in Saudi Arabia :).

**Comment by SanderGit** [comment]
I understand that an [API call is needed for this](https://developers.google.com/safe-browsing) so it cannot be fixed with DNS only - would definitely be a feature request to consider. The only thing that would come near I could think of is to subscribe to a block list that provides similar functionality.

**Comment by ameshkov** [comment]
This function would also require the user to specify an API key that'll allow using Google Safebrowsing.

**Comment by valaria247** [comment]
> This function would also require the user to specify an API key that'll allow using Google Safebrowsing.

@ameshkov Wait! Isn't it possible for adguard to implement something instead integrate this?

@ai-utt It looks like your country also block adguard☹️ and welcome for google?

**Comment by ameshkov** [comment]
> @ameshkov Wait! Isn't it possible for adguard to implement something instead integrate this?

Well, we're not implementing anything yet:) Also, even if we do, this won't be enabled by default.

**Comment by ameshkov** [comment]
We may also consider providing an option Yandex Safebrowsing support as well: https://github.com/AdguardTeam/AdGuardHome/issues/1952

Again, note that this feature would require the user to provide their own API keys to these services.

**Comment by AngelFalse** [comment]
How does Google SafeBrowsing actually work? Does it provides the same level of privacy as the AdGuard Browsing security web service?

**Comment by michaelb-ae** [comment]
i would ask if adguard can do it with their own google api key, because, some (synology) asked us to provide one (even if adguard generate it in our name), because i don't want to create a google account only to generate an api key.
Thanks ^^

**Comment by Aikatsui** [comment]
Source @ZeroClover https://github.com/AdguardTeam/AdGuardHome/issues/2857

>a similar pre-cache can be implemented by AdGuard.
>For the following reasons, I think the current implementation of safe browsing is not good and could probably be improved to enhance performance and privacy.

>The Google Safe Browsing Update API allows clients to perform initial checks locally by enabling them to download a list of safe browsing hashes in advance. Only if they match will they need to send further requests to the server to perform the checks. This not only makes processing much faster but also improves privacy. The server does not know all of the client's browsing history.
>Besides, even if you do not use Google-related services for privacy reasons, you should consider other implementations than sending a request at every query.

**Comment by Aikatsui** [comment]
It seems if adguard implement alternative to google safe browsing. it'll not get blocked anywhere in world ;)
